### PR TITLE
Add instructions for VPA release

### PR DESCRIPTION
**Pull request by jbartosik**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Add step-by-step instructions for VPA release process.

### CODE REVIEW

File: vertical-pod-autoscaler/RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+1. [ ] Open issue to track the release.
+2. [ ] Update VPA version const.
+3. [ ] Build and stage images.
+4. [ ] Test the release.
+5. [ ] Promote image.
+6. [ ] Finalize release.
+
+## Open issue to track the release
+
+We use the issue to communicate what is state of the release.
```
Comment (kgolab) [inline vertical-pod-autoscaler/RELEASE.md:19]: Do we have a template for the release tracking issue?
If not, maybe we can add it together with the next release or even inside this PR?

File: vertical-pod-autoscaler/RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    [common/version-go](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/common/version.go)
+    to `0.${next-minor}.0`,
+5. [ ] Commit and merge the change,
+6. [ ] Go to the merged change,
+7. [ ] [Create a new branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository) named `vpa-release-0.${next-minor}` from the
+    merged change.
+
+### New patch release
+
+1.  [ ] Bump the patch version number in VerticalPodAutoscalerVersion constant in
```
Comment (kgolab) [inline vertical-pod-autoscaler/RELEASE.md:39]: Bit nit-picking but I hope that steps 1-3 from new minor release also apply here, right?

File: vertical-pod-autoscaler/RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
+# VPA Release Instructions
+
+These are instructions for releasing VPA. We aim to release a new VPA minor version after each minor Kubernetes release.
+We release patch versions as needed.
+
+Before doing the release for the first time check if you have all the necessary permissions.
```
Comment (kgolab) [inline vertical-pod-autoscaler/RELEASE.md:6]: How to do this / where to find the list of required permissions?

**Comment by kgolab** [comment]
Thanks for creating the instructions, it's really good to have the process well-documented.

File: vertical-pod-autoscaler/RELEASE.md (hunk 1 above)
Comment (jbartosik) [reply vertical-pod-autoscaler/RELEASE.md:19]: No, we don't have a template. For now I've only opened an issue, called it something like "Release VPA 0.X.X" and linked all info that seemed relevant (for example #5355).

I opened #5454 to add a template

File: vertical-pod-autoscaler/RELEASE.md (hunk 2 above)
Comment (jbartosik) [reply vertical-pod-autoscaler/RELEASE.md:41]: Yes, I moved those steps a level up

**Comment by jbartosik** [comment]
@kgolab Thanks, please take another look

**Comment by kgolab** [comment]
Thanks!

/lgtm

**Pull request merged**